What is a smart contract

jamesconnolly93 opened the pull request:
Added a more robust, but beginner-friendly definition of smart contracts/decentral applications to the "What is Ethereum" page. I believe this will be helpful in allowing users to understand the usefulness of Ethereum beyond legal contracts.

ChazSchmidt commented on docs/ethereum-basics/what-is-ethereum.md:
```diff
@@ -11,6 +11,33 @@ Ethereum is an open-source, public, blockchain-based distributed ledger featurin
+The term "Smart Contract" was coined by Nick Szabo in the 90's. Szabo used the basic example of a vending machine to describe how real-world contractual obligations can be programmed into software and hardware systems. Everyone who puts the correct amount of coins into the machine can expect to receive a product in exchange. Similarly, on Ethereum, contracts can hold value and unlock it only if s ...
+
+The scripting language used by Ethereum is Turing-complete, essentially meaning that the types of DApps users can design is limited only by their imagination. The [Ethereum Whitepaper] (https://github.com/ethereum/wiki/wiki/White-Paper) divides DApps into three categories:
+
```
I think everything is good until this breakdown of categories. I understand idea of giving examples but I think to a newcomer, these categories seem official. And sometimes, the lines aren't so clear. Like for example, a DAOs can be for-profit entities and collectibles typically are associated with value otherwise why collect them.

jamesconnolly93 replied:
Ok I see where you're coming from here. I'll update my pull request to omit that part.

jamesconnolly93 commented:
@ChazSchmidt Per your recommendation, I've removed the categories/examples section. I also added a link to the Ethereum whitepaper to the "Resources" section. Seems like a good thing to have on a "What is Ethereum?" page given that it's basically the canonical text on the topic.
